src/app/brew/brew-add/brew-add.component.ts

```diff
@@ -34,25 +30,29 @@ import BREW_TRACKING from '../../../data/tracking/brewTracking';
 import { UIAnalytics } from '../../../services/uiAnalytics';
 
 import { SettingsPopoverBluetoothActionsComponent } from '../../settings/settings-popover-bluetooth-actions/settings-popover-bluetooth-actions.component';
-import { BluetoothScale, SCALE_TIMER_COMMAND } from '../../../classes/devices';
+import { PreparationDeviceType } from '../../../classes/preparationDevice';
+import { UIHelper } from '../../../services/uiHelper';
+import {
+  BluetoothScale,
+  SCALE_TIMER_COMMAND,
+  sleep,
+} from '../../../classes/devices';
 import {
   CoffeeBluetoothDevicesService,
   CoffeeBluetoothServiceEvent,
 } from '../../../services/coffeeBluetoothDevices/coffee-bluetooth-devices.service';
-import { PreparationDeviceType } from '../../../classes/preparationDevice';
-import { UIHelper } from '../../../services/uiHelper';
 import { VisualizerService } from '../../../services/visualizerService/visualizer-service.service';
 import { Subscription } from 'rxjs';
 
 declare var Plotly;
-declare var window;
+
 @Component({
   selector: 'brew-add',
   templateUrl: './brew-add.component.html',
   styleUrls: ['./brew-add.component.scss'],
 })
-export class BrewAddComponent implements OnInit {
-  public static COMPONENT_ID: string = 'brew-add';
+export class BrewAddComponent implements OnInit, OnDestroy {
```

Comment:
Adding this `OnDestroy` is a good bug fix (`ngOnDestroy` was implemented but not being called because `BrewAddComponent` didn't implement OnDestroy) but it's a behaviour change which should probably be done in a separate change from fixing the tests?

Reply:
https://github.com/graphefruit/Beanconqueror/pull/738 for the OnDestroy change so it can be merged separately.

Reply:
I've merged #738 so we can go on here